File: HPWorkpath_v1.6.3/Samples/ExampleExtensions/source/GoogleSigninSample/src/com/hp/workpath/sample/googlesigninsample/model/AccountInfo.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.googlesigninsample.model;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;

public class AccountInfo {

    @SerializedName("access_token")
    String accessToken;

    @SerializedName("expires_in")
    int expiresIn;

    @SerializedName("id_token")
    String idToken;

    @SerializedName("refresh_token")
    String refreshToken;

    @SerializedName("scope")
    String scope;

    @SerializedName("token_type")
    String tokenType;

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public int getExpiresIn() {
        return expiresIn;
    }

    public String getIdToken() {
        return idToken;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public String getTokenType() {
        return tokenType;
    }

    public String getScope() {
        return scope;
    }

    @Override
    public String toString() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleExtensions/source/GoogleSigninSample/src/com/hp/workpath/sample/googlesigninsample/model/AccountInfo.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.googlesigninsample.model

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

class AccountInfo {
    @SerializedName("access_token")
    var accessToken: String? = null

    @SerializedName("expires_in")
    var expiresIn = 0

    @SerializedName("id_token")
    var idToken: String? = null

    @SerializedName("refresh_token")
    var refreshToken: String? = null

    @SerializedName("scope")
    var scope: String? = null

    @SerializedName("token_type")
    var tokenType: String? = null
    override fun toString(): String {
        val gson = Gson()
        return gson.toJson(this)
    }
}